### PR TITLE
Add Tensorflow stubs

### PR DESCRIPTION
**Pull request by MightyPiggie**

### CODE REVIEW

File: stubs/tensorflow/tensorflow/keras/layers/__init__.pyi [hunk 1]
```diff
@@ -442,4 +442,21 @@ class GaussianDropout(Layer[tf.Tensor, tf.Tensor]):
         name: str | None = None,
     ) -> None: ...
 
+class MaxPooling2D(Layer[tf.Tensor, tf.Tensor]):
+    def __init__(
+        self,
+        pool_size: int | tuple[int, int] = (2, 2),
+        strides: int | tuple[int, int] | None = None,
+        padding: Literal["valid", "same"] = "valid",
```
Comment (yangdanny97) [inline stubs/tensorflow/tensorflow/keras/layers/__init__.pyi:450]: We might not be able to type this as a literal, the docs mention it's case-insensitive

https://www.tensorflow.org/api_docs/python/tf/keras/layers/MaxPool2D#args

there's examples of code that pass the value in all caps

https://github.com/search?q=padding%3D%22valid%22+language%3APython+MaxPool2D&type=code

File: stubs/tensorflow/tensorflow/keras/layers/__init__.pyi [hunk 2]
```diff
@@ -442,4 +442,21 @@ class GaussianDropout(Layer[tf.Tensor, tf.Tensor]):
...
+        data_format: None | Literal["channels_last", "channels_first"] = None,
```
Comment (yangdanny97) [inline stubs/tensorflow/tensorflow/keras/layers/__init__.pyi:451]: `Literal["channels_last", "channels_first"] | None` to match the style of the other optionals.

File: stubs/tensorflow/tensorflow/keras/layers/__init__.pyi [hunk 3]
```diff
@@ -442,4 +442,21 @@ class GaussianDropout(Layer[tf.Tensor, tf.Tensor]):
...
+        name: str | None = None,
```
Comment (yangdanny97) [inline stubs/tensorflow/tensorflow/keras/layers/__init__.pyi:452]: The implementation seems to accept `**kwargs` though I'm not sure the purpose

https://github.com/keras-team/keras/blob/v3.3.3/keras/src/layers/pooling/max_pooling2d.py#L96

File: stubs/tensorflow/tensorflow/keras/layers/__init__.pyi [hunk 4]
```diff
@@ -442,4 +442,21 @@ class GaussianDropout(Layer[tf.Tensor, tf.Tensor]):
         name: str | None = None,
     ) -> None: ...
 
+class MaxPooling2D(Layer[tf.Tensor, tf.Tensor]):
```
Comment (yangdanny97) [inline stubs/tensorflow/tensorflow/keras/layers/__init__.pyi:445]: In the implementation this doesn't extend Layer directly, it extends BasePooling

https://github.com/keras-team/keras/blob/v3.3.3/keras/src/layers/pooling/base_pooling.py

File: stubs/tensorflow/tensorflow/keras/layers/__init__.pyi [hunk 5]
```diff
@@ -442,4 +442,21 @@ class GaussianDropout(Layer[tf.Tensor, tf.Tensor]):
...
+        pool_size: int | tuple[int, int] = (2, 2),
+        strides: int | tuple[int, int] | None = None,
+        padding: Literal["valid", "same"] = "valid",
+        data_format: None | Literal["channels_last", "channels_first"] = None,
+        name: str | None = None,
+    ) -> None: ...
+
+MaxPool2D = MaxPooling2D
+
+class GlobalAveragePooling2D(Layer[tf.Tensor, tf.Tensor]):
```
Comment (yangdanny97) [inline stubs/tensorflow/tensorflow/keras/layers/__init__.pyi:457]: Similar comments as above re: ordering of `| None`, and the base class here is `BaseGlobalPooling`
